app/scripts/controllers/preferences.test.js

```diff
@@ -100,6 +101,45 @@ describe('preferences controller', function () {
     });
   });
 
+  describe('updateTokenType', function () {
+    it('should add isERC721 = true to token object in state when token is collectible and in our contract-metadata repo', async function () {
+      const contractAddresses = Object.keys(contractMaps);
+      const erc721ContractAddresses = contractAddresses.filter(
+        (contractAddress) => contractMaps[contractAddress].erc721 === true,
+      );
+      const address = erc721ContractAddresses[0];
+      const { symbol, decimals } = contractMaps[address];
+      preferencesController.store.updateState({
+        tokens: [{ address, symbol, decimals }],
+      });
+      const result = await preferencesController.updateTokenType(address);
+      assert.equal(result.isERC721, true);
+    });
+
+    it('should add isERC721 = true to token object in state when token is collectible and not in our contract-metadata repo', async function () {
+      preferencesController.store.updateState({
+        tokens: [
+          {
+            address: '0xda5584cc586d07c7141aa427224a4bd58e64af7d',
+            symbol: 'TESTNFT',
+            decimals: '0',
+          },
+        ],
+      });
+      const sandbox = sinon.createSandbox();
+      sandbox
+        .stub(preferencesController, '_detectIsERC721')
+        .callsFake(() => true);
+      // sandbox.stub(ethers, 'Contract');
+      // sinon.stub(ethers.Contract.prototype, 'supportsInterface').returns(true);
```

Comment:
@danjm @brad-decker do either of you happen to know how to stub this method correctly? I've poured through all related content on stackoverflow but can't seem to figure it out.

Reply:
so this is a method on the object/class returend by the call to

```
new ethers.Contract(
      tokenAddress,
      abiERC721,
      this.ethersProvider,
    )
```

Reply:
So first I would just validate that `ethers.Contract` is called with the expected parameters.

Reply:
but then I think you would need to stub `ethers.Contract`, have it return an object with a `supportsInterface` method, verify that method gets called with the expected param, and stub `supportsInterface` to return true

Reply:
Also, you may need to stub the constructor. So maybe something like:
```
const supportsInterfaceStub = sandbox.stub().returns(true);
sandbox.stub(ethers.Contract, 'constructor')
    .callsFake(() => ({ supportsInterface: supportsInterfaceStub }))
```

Reply:
Please do not introduce any more usages of `sinon.sandbox()` :pray: I've been slowly trying to get rid of them everywhere. They just make sinon more difficult to use. They're only meant for obscure cases where you need to manage multiple pools of distinct stubs simultaneously. The default sandbox works for everything we do.